src/tls13.c

```diff
@@ -8218,6 +8218,85 @@ int wolfSSL_request_certificate(WOLFSSL* ssl)
 }
 #endif /* !NO_CERTS && WOLFSSL_POST_HANDSHAKE_AUTH */
 
+
+#if defined(OPENSSL_EXTRA) && defined(HAVE_ECC)
+static int populate_groups(int* groups, int max_count, char *list)
+{
+    char *end;
+    size_t len;
+    int count = 0;
+    const WOLF_EC_NIST_NAME* nist_name;
+
+    if (!groups || !list) {
+        return -1;
+    }
+
+    for (end = list; ; list = ++end) {
+        if (count > max_count) {
+            WOLFSSL_MSG("Too many curves in list");
+            return -1;
+        }
+        while (*end != ':' && *end != '\0') end++;
+        len = end - list; /* end points to char after end
+                           * of curve name so no need for -1 */
+        if ((len < kNistCurves_MIN_NAME_LEN) ||
+                (len > kNistCurves_MAX_NAME_LEN)) {
+            WOLFSSL_MSG("Unrecognized curve name in list");
+            return -1;
+        }
+        for (nist_name = kNistCurves; nist_name->name != NULL; nist_name++) {
+            if (XSTRNCMP(list, nist_name->name, nist_name->name_len) == 0) {
+                break;
+            }
+        }
+        if (!nist_name->name_len) {
```

Comment:
The for loop above uses `nist_name->name != NULL` and this uses `!nist_name->name_len`. Was that on purpose?

Reply:
You are right. This was a mistake.